### PR TITLE
Sos 250 add a confirmation dialog when accepting rejecting shares or updating permissions

### PR DESCRIPTION
**Pull request by SeanTConrad**
Please see the test plan here: https://www.notion.so/Permissions-e32418ecef5649729868d35eef5b9eb4?pvs=4#51f4ba80fb0043f18ab4342e0e8a61be

List of changes:

1. Uninstalled "material-ui-snackbar-provider" because it sucked. No configurability. Didn't help create resusable code.
2. Fix to only show unarchived and unaccepted messages in Notifications menu in TopAppBar
3. Fixed a merge error where CompanyCard was missing a line of code to show the widget from the config
4. Created a generic, reusable Snackbar component
5. On the Sharing Page:
6. Messages only shows for companies with receiving enabled
7. Existing shares now show in the permissions grid so they can be disabled
8. Current company removed from list of receiving companies (no need to share with yourself)
9. Permissions based on Owner or Employee
10. Confirmation dialog for sharing
11. Dynamic enabling / highlighting as user goes through steps of sharing
12. Some copy is configurable, e.g. for CreditChain "Share your information with lenders"
13. Fixed spelling error in API service "permisssions"
14. API to revoke company to company permissions added pointing to ATD
15. Fixed bug in messages service where modified messages ended up twice in the messages list

### CODE REVIEW

File: client/src/components/Alerts.tsx [hunk 1]
```diff
@@ -39,21 +44,29 @@ export default function Alerts() {
         setAnchorEl(null);
     };
 
+    const id = open ? 'message-popover' : undefined;
+
+    let messageComponents;
+
+    const unreadCount = filteredMessages.filter((m)=>m.is_read !== true).length;
```
Comment (aylusltd) [inline client/src/components/Alerts.tsx:51]: Should do && !message.archived no?

File: client/src/components/SOSSnackbar.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// const permUpdated:ISOSSnackbarProps = {
+//     variant: snackbarVariants.success,
+//     message: 'Permissions updated.',
+//     setOpen: () => setSnackbar({open:false}),
+//     open: true
+// }
+// setSnackbar(permUpdated);
+
+import * as React from 'react';
+import Snackbar from '@mui/material/Snackbar';
```
Comment (aylusltd) [inline client/src/components/SOSSnackbar.tsx:16]: Didn't you remove this?

File: client/src/components/index.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ export { default as OAuthLink } from './OAuthLink';
 export { default as ReportHeader } from './ReportHeader';
 export { default as SectionHeader} from './SectionHeader';
 export { default as Sockets } from './Sockets';
+export { default as SOSSnackbar } from './SOSSnackbar';
```
Comment (aylusltd) [inline client/src/components/index.ts:35]: Same?

File: client/src/services/messages.tsx [hunk 4]
```diff
@@ -151,6 +151,7 @@ const MessagesContext = createContext<MessagesContextShape>(
         try {
             const { data: payload } = await apiArchiveMessage(messageId);
             if (payload != null) {
+                console.log("payload", payload)
```
Comment (aylusltd) [inline client/src/services/messages.tsx:154]: Should remove before merge

File: client/src/components/Alerts.tsx (hunk 1 above)
Comment (SeanTConrad) [reply client/src/components/Alerts.tsx:51]: filteredMessages was already filtered for !archived above.
`filteredMessages = messageState.messages.filter((c)=>{
    return !c.archived && !c.approved
})`

File: client/src/services/messages.tsx (hunk 4 above)
Comment (SeanTConrad) [reply client/src/services/messages.tsx]: Done.

**Review by aylusltd — APPROVED** [review]

**Pull request merged**